wifiphisher/interfaces.py

```diff
@@ -0,0 +1,440 @@
+    if down_error:
+        raise SetMonitorModeError(down_error)
+
+    # set the interface to managed mode
+    iw_command = subprocess.Popen(["iwconfig", interface, "mode", mode],
+                                  stdout=DEVNULL, stderr=subprocess.PIPE)
+
+    # get any possible error
+    iw_error = iw_command.communicate()[1]
+
+    # raise an error if any errors were produced
+    if iw_error:
+        raise SetMonitorModeError(down_error)
+
+    # turn the interface on
+    on_command = subprocess.Popen(["ifconfig", interface, "up"],
+                                  stdout=DEVNULL, stderr=subprocess.PIPE)
+
+    # get any possible error
+    on_error = on_command.communicate()[1]
+
+    # raise an error if any errors were produced
+    if on_error:
+        raise SetMonitorModeError(on_error)
+
+
+def get_wireless_interfaces():
+    """
+    Return a list of available wireless interfaces.
+
+    Args:
+        None.
+
+    Returns:
+        (list): A list of available wireless interfaces.
+    """
+
+    # Initialize a list to store the wireless interfaces
+    wireless_interfaces = list()
+
+    # run the command to get the interfaces
+    iwconfig_command = subprocess.Popen("iwconfig", stdout=subprocess.PIPE,
+                                        stderr=DEVNULL)
+
+    # add all the wireless interfaces to the list
+    for line in iwconfig_command.communicate()[0].split("\n"):
+        # check if the line includes an interface
+        if len(line) > 0 and line[0] != " ":
+            # add the interface to the list if it is wireless
+            if line.startswith("w"):
+                wireless_interfaces.append(line[:line.find(" ")])
+
+    return wireless_interfaces
+
+
+def get_interfaces(jamming_argument, ap_argument):
+    """
+    Return a tuple containing an interface with monitor mode fallowed by an
+    interface with AP mode available.
+
+    Args:
+        jamming_argument (str): The value for jamming interface
+                                argument (-jI).
+
+        ap_argument      (str): The value for ap interface argument (-aI).
+
+    Returns:
+        (str, str): A tuple containing monitor interface fallowed by
+                    ap interface.
+
+    Raises:
+        NotEnoughInterfacesFound: If less than two interfaces is
+                                  discovered.
+
+        JammingInterfaceInvalid : If an invalid jamming interface (-jI) is
+                                 supplied.
+
+        ApInterfaceInvalid      : If an invalid ap interface (-aI) is
+                                  supplied.
+
+        NoApInterfaceFound      : In case of not finding a valid AP
+                                  interface.
+
+        NoMonitorInterfaceFound : In case of not finding a valid monitor
+                                 interface.
+
+    Dependencies:
+        get_wireless_interfaces
+        check_compatibility
+        get_interface
+        select_interface_automatically
```

Comment:
I wrote this to serve as a warning to anyone who might change one of those functions since ```get_interfaces``` function is dependent on these functions. Maybe ```dependent functions``` would be a better name for it.